src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.xml

```diff
@@ -110,5 +110,16 @@
             <argument type="service" id="router.request_context" on-invalid="ignore" />
             <argument type="service" id="logger" on-invalid="ignore" />
         </service>
+
+        <service id="Symfony\Bundle\FrameworkBundle\Controller\RedirectController" public="true">
+            <argument type="service" id="router" />
+            <argument>%request_listener.http_port%</argument>
+            <argument>%request_listener.https_port%</argument>
+        </service>
+
+        <service id="Symfony\Bundle\FrameworkBundle\Controller\TemplateController" public="true">
```

Comment:
I don't think this service belongs in routing.xml. It has nothing to do with routing?!

Reply:
These services are useful only when a route references them, that's why I made it this way.
Did I miss something else?

Reply:
Can't one use them without routing using fragments?